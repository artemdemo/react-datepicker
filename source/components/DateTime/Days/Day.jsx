import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { propIsMoment } from '../propTypes';

import './Day.less';

class Day extends React.Component {
    clickHandler(validDate) {
        const { date, onClick } = this.props;
        if (validDate) {
            onClick && onClick(date);
        }
    }

    render() {
        const { date, current, selected, faded, isValidDate } = this.props;
        const dayClass = classnames({
            'datetime-day': true,
        });
        const validDate = isValidDate ? isValidDate(date) : true;
        const contentClass = classnames({
            'datetime-day__content': true,
            'datetime-day__content_current': current,
            'datetime-day__content_selected': selected,
            'datetime-day__content_faded': faded,
            'datetime-day__content_disabled': !validDate,
        });
        return (
            <td
                className={dayClass}
                onClick={this.clickHandler.bind(this, validDate)}
            >
                <div className={contentClass}>
                    {date.format('D')}
                </div>
            </td>
        );
    }
}

Day.displayName = 'Day';

Day.propTypes = {
    date: propIsMoment.isRequired,
    onClick: PropTypes.func,
    isValidDate: PropTypes.func,
    current: PropTypes.bool,
    selected: PropTypes.bool,
    faded: PropTypes.bool,
};

Day.defaultProps = {
    onClick: null,
    isValidDate: null,
    current: false,
    selected: false,
    faded: false,
};

export default Day;
