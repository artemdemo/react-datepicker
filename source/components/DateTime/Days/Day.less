@import '../styles/general';

.datetime-day {
    text-align: center;
    cursor: pointer;
}
    .datetime-day__content {
        .clickable();

        border: 2px solid transparent;
        padding: 0 3px;
        background-color: transparent;
    }
        .datetime-day__content_current {
            border-color: @hover-color;

            &.datetime-day__content_selected {
                border-color: darken(@hover-color, 15%);
            }
        }
        .datetime-day__content_selected {
            background-color: @hover-color;
        }
        .datetime-day__content_faded {
            color: fade(@text-color, 35%);
        }
        .datetime-day__content_disabled {
            color: fade(@text-color, 15%);
            cursor: not-allowed;

            &:hover {
                background: transparent;
            }
        }
