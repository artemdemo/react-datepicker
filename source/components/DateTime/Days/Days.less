@import '../styles/general';

.datetime-days {
    .datetime-table();

    width: auto;
}

