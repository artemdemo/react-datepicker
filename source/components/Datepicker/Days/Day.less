@import '../datepicker-general';

.datepicker-day {
    text-align: center;
    cursor: pointer;
}
    .datepicker-day__content {
        border: 2px solid transparent;
        border-radius: 4px;
        padding: 3px;

        &:hover {
            background: @hover-color;
        }
    }
        .datepicker-day__content_current {
            border-color: @hover-color;
        }
        .datepicker-day__content_faded {
            color: fade(@text-color, 30%);
        }
