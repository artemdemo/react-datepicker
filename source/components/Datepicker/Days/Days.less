@import '../datepicker-general';

.datepicker-days {
    .datepicker-table();
}

