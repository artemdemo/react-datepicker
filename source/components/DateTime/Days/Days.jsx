import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Day from './Day';
import TitleController from '../TitleController/TitleController';
import Separator from '../Separator/Separator';

import './Days.less';

class Days extends React.Component {
    constructor(props) {
        super(props);
        const { date } = this.props;

        this.state = {
            selectedDate: date.clone(),
        };
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.date !== this.props.date) {
            this.setState({
                selectedDate: nextProps.date.clone(),
            });
        }
    }

    getDaysOfWeek() {
        const localeDate = this.state.selectedDate.localeData();
        const weekdaysMin = localeDate.weekdaysMin();
        const first = localeDate.firstDayOfWeek();
        const dow = [];
        let i = 0;

        weekdaysMin.forEach((day) => {
            dow[((7 + (i++)) - first) % 7] = day;
        });

        return dow;
    }

    dayClickHandler(date) {
        const { onDateChange } = this.props;
        onDateChange(date);
    }

    changeMonth(newSelectedDate) {
        this.setState({
            selectedDate: newSelectedDate,
        });
    }

    renderDaysOfWeek() {
        return (
            <thead>
                <tr>
                    {this.getDaysOfWeek().map((day, index) => (
                        <th key={`datetime-day-of-week-${index}`}>
                            {day}
                        </th>
                    ))}
                </tr>
            </thead>
        );
    }

    renderDays() {
        const selectedDate = this.state.selectedDate;
        const prevMonth = selectedDate.clone().subtract(1, 'month');
        const currentYear = selectedDate.year();
        const currentMonth = selectedDate.month();
        const weeksRows = [];
        let daysColumns = [];

        // Go to the last week of the previous month
        prevMonth.date(prevMonth.daysInMonth()).startOf('week');
        const lastDay = prevMonth.clone().add(42, 'day');

        // resetting date will clear time data
        prevMonth.set({
            hour: selectedDate.get('hour'),
            minute: selectedDate.get('minute'),
            second: selectedDate.get('second'),
            millisecond: selectedDate.get('millisecond'),
        });

        const { isValidDate } = this.props;

        while (prevMonth.isBefore(lastDay)) {
            const pastMonth = (prevMonth.year() === currentYear && prevMonth.month() < currentMonth) ||
                               prevMonth.year() < currentYear;

            const futureMonth = (prevMonth.year() === currentYear && prevMonth.month() > currentMonth) ||
                                 prevMonth.year() > currentYear;

            daysColumns.push(
                <Day
                    date={prevMonth.clone()}
                    onClick={this.dayClickHandler.bind(this)}
                    current={prevMonth.isSame(moment(), 'day')}
                    selected={prevMonth.isSame(selectedDate)}
                    faded={pastMonth || futureMonth}
                    isValidDate={isValidDate}
                    key={`days-columns-${prevMonth.format('M_D')}`}>
                    {prevMonth.format('D')}
                </Day>,
            );

            if (daysColumns.length === 7) {
                weeksRows.push(
                    <tr key={`weeks-rows-${prevMonth.format('M_D')}`}>
                        {daysColumns}
                    </tr>,
                );
                daysColumns = [];
            }

            prevMonth.add(1, 'day');
        }

        return (
            <tbody>
                {weeksRows}
            </tbody>
        );
    }

    render() {
        return (
            <div>
                <TitleController
                    date={this.state.selectedDate}
                    onChange={this.changeMonth.bind(this)} />
                <Separator />
                <table
                    className='datetime-days'>
                    {this.renderDaysOfWeek()}
                    {this.renderDays()}
                </table>
            </div>
        );
    }
}

Days.displayName = 'Days';

Days.propTypes = {
    date: PropTypes.instanceOf(moment).isRequired,
    onDateChange: PropTypes.func.isRequired,
    isValidDate: PropTypes.func,
};

Days.defaultProps = {
    isValidDate: null,
};

export default Days;
