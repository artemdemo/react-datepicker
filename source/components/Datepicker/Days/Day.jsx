import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Day.less';

class Day extends React.Component {
    clickHandler() {
        const { date, onClick } = this.props;
        onClick && onClick(date);
    }

    render() {
        const { date, current, faded } = this.props;
        const dayClass = classnames({
            'datepicker-day': true,
        });
        const contentClass = classnames({
            'datepicker-day__content': true,
            'datepicker-day__content_current': current,
            'datepicker-day__content_faded': faded,
        });
        return (
            <td
                className={dayClass}
                onClick={this.clickHandler.bind(this)}>
                <div className={contentClass}>
                    {date.format('D')}
                </div>
            </td>
        );
    }
}

Day.propTypes = {
    date: PropTypes.shape({}).isRequired,
    onClick: PropTypes.func,
    current: PropTypes.bool,
    faded: PropTypes.bool,
};

Day.defaultProps = {
    onClick: null,
    current: false,
    faded: false,
};

export default Day;
