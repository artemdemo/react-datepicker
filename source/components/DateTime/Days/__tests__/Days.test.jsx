import React from 'react';
import moment from 'moment';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Days from '../Days';

describe('<Days />', () => {
    const date = moment(1502459094000);

    it('simple render', () => {
        const tree = renderer.create(
            <Days
                date={date}
                selectedDate={date}
                onDateChange={() => {}}
                test
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('handle click on date', () => {
        let selectedTimestamp = 0;
        const wrapper = mount(
            <Days
                date={date}
                selectedDate={date}
                onDateChange={(date) => {
                    selectedTimestamp = date.valueOf();
                }}
                test
            />,
        );
        wrapper.find('.datetime-day').first().simulate('click');
        expect(selectedTimestamp).toEqual(1501422294000);
    });

    it('all dates after 1 of august (not incl) should be disabled', () => {
        const tree = renderer.create(
            <Days
                date={date}
                selectedDate={date}
                isValidDate={_date => _date.valueOf() < 1501681494000}
                onDateChange={() => {}}
                test
            />,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
