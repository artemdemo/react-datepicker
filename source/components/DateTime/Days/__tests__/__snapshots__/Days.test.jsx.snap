// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Days /> all dates after 1 of august (not incl) should be disabled 1`] = `
<table
  className="datetime-days"
>
  <thead>
    <tr>
      <th>
        Su
      </th>
      <th>
        Mo
      </th>
      <th>
        Tu
      </th>
      <th>
        We
      </th>
      <th>
        Th
      </th>
      <th>
        Fr
      </th>
      <th>
        Sa
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          30
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          31
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          1
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          2
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          3
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          4
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          5
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          6
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          7
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          8
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          9
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          10
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_current datetime-day__content_selected datetime-day__content_disabled"
        >
          11
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          12
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          13
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          14
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          15
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          16
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          17
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          18
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          19
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          20
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          21
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          22
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          23
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          24
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          25
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          26
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          27
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          28
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          29
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          30
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_disabled"
        >
          31
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          1
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          2
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          3
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          4
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          5
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          6
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          7
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          8
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded datetime-day__content_disabled"
        >
          9
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Days /> simple render 1`] = `
<table
  className="datetime-days"
>
  <thead>
    <tr>
      <th>
        Su
      </th>
      <th>
        Mo
      </th>
      <th>
        Tu
      </th>
      <th>
        We
      </th>
      <th>
        Th
      </th>
      <th>
        Fr
      </th>
      <th>
        Sa
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          30
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          31
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          1
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          2
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          3
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          4
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          5
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          6
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          7
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          8
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          9
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          10
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_current datetime-day__content_selected"
        >
          11
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          12
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          13
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          14
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          15
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          16
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          17
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          18
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          19
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          20
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          21
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          22
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          23
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          24
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          25
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          26
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          27
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          28
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          29
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          30
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content"
        >
          31
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          1
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          2
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          3
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          4
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          5
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          6
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          7
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          8
        </div>
      </td>
      <td
        className="datetime-day"
        onClick={[Function]}
      >
        <div
          className="datetime-day__content datetime-day__content_faded"
        >
          9
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
